docs-search/src/Docs/Search/JsonCodec.purs

```diff
@@ -0,0 +1,51 @@
+-- | Utilities to interop between different codec libraries
+module Docs.Search.JsonCodec
+  ( fromGeneric
+  , fromJsonUnidirectional
+  , inject
+  ) where
+
+import Prelude
+import Prim.Row (class Cons)
+
+import Codec.Json.Unidirectional.Value (DecodeError, printDecodeError)
+import Data.Argonaut.Core (Json, fromString, stringify)
+import Data.Argonaut.Decode (class DecodeJson, decodeJson)
+import Data.Argonaut.Decode.Error as Generic
+import Data.Argonaut.Encode (class EncodeJson, encodeJson)
+import Data.Codec.Argonaut (JsonCodec, JsonDecodeError(..))
+import Data.Codec.Argonaut.Common as CA
+import Data.Codec.Argonaut.Sum as CAS
+import Data.Codec.Argonaut.Record as CAR
+import Data.Either (Either(..))
+import Data.Profunctor.Choice (left)
+import Data.Symbol (class IsSymbol)
+import Data.Variant
+import Type.Proxy (Proxy(..))
+
+-- | Equivalent to `Data.Variant.inj`, just uses a Visible Type Application instead of Proxy
+-- | Useful for deriving codecs for sum types
+inject :: forall @sym a r1 r2. Cons sym a r1 r2 => IsSymbol sym => a -> Variant r2
+inject = inj (Proxy :: _ sym)
+
+fromJsonUnidirectional
+  :: forall a
+   . (Json -> Either DecodeError a)
+  -> (Json -> Either JsonDecodeError a)
+fromJsonUnidirectional = map $ left convertError
+  where
+  convertError :: DecodeError -> JsonDecodeError
+  convertError = printDecodeError >>> TypeMismatch
```

Comment:
I'm cheating here and using `TypeMismatch` as a custom error type, to easily convert from a `json-codecs` error into a `codec-argonaut` error.

@JordanMartinez @f-f Thoughts?

Reply:
The more I think about this, I don't think this deviates too far from the intended usage of `TypeMismatch`.

Reply:
Whenever I need to use a custom message in codec-argonaut/argonaut-codec, I use `UnexpectedValue $ encodeJson "Here's my custom message"`.